domain/src/SharePooling/Services/ReversionFinder.php

```diff
@@ -0,0 +1,120 @@
+<?php
+
+namespace Domain\SharePooling\Services;
+
+use Brick\DateTime\LocalDate;
+use Domain\SharePooling\Actions\AcquireSharePoolingToken;
+use Domain\SharePooling\Actions\DisposeOfSharePoolingToken;
+use Domain\SharePooling\ValueObjects\SharePoolingTokenDisposals;
+use Domain\SharePooling\ValueObjects\SharePoolingTransactions;
+use Domain\ValueObjects\Quantity;
+
+final class ReversionFinder
```

Comment:
ℹ️ This service identifies and returns the disposals that need to be reverted upon a new acquisition or disposal